### PR TITLE
Making a possibility to hide a trending line in a cost insights plugin

### PR DESCRIPTION
**Pull request by acierto**
Signed-off-by: bnechyporenko <bnechyporenko@bol.com>

## Hey, I just made a Pull Request!

Cost Insight plugin in `CostOverviewChart` has an issue with the alignment of a trending line and a "cost trend" calculation.
As for trending line an approximation used, but for cost trend is the difference between edge points. 
That issue is leading to a confusing chart. 
Like this one:
image-2022-11-15-09-33-31-468
So as we can see here, cost trend is -3% but the trending line is growing. 

To not do significant changes, we would like to be able to hide the trending line. 
It doesn't bring much of the value and also not consistent UX wise with "Breakdown by product" tab.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/cost-insights/src/components/CostOverviewCard/CostOverviewCard.test.tsx [hunk 1]
```diff
@@ -38,13 +39,30 @@ const mockGroupDailyCost: Cost = {
   trendline: trendlineOf(MockAggregatedDailyCosts),
 };
 
+type TestInputPluginOptions = {
+  showTrendLine: boolean;
```
Comment (jhaals) [inline plugins/cost-insights/src/components/CostOverviewCard/CostOverviewCard.test.tsx:43]: isn't it nicer to make it `hideTrendLine?: boolean;` so it is both optional and defaults to false.

File: plugins/cost-insights/src/components/CostOverviewCard/CostOverviewCard.test.tsx [hunk 2]
```diff
@@ -38,13 +39,30 @@ const mockGroupDailyCost: Cost = {
...
+};
+
+type TestPluginOptions = {
+  showTrendLine: boolean;
+};
+
+const plugin = createPlugin({
```
Comment (jhaals) [inline plugins/cost-insights/src/components/CostOverviewCard/CostOverviewCard.test.tsx:50]: We probably need some form of test util to reduce the boilerplate to test this functionality, probably good to add that before as a separate PR.

**Review by jhaals — CHANGES_REQUESTED** [review]
Thanks for the PR! I've added some suggestions

File: plugins/cost-insights/src/components/CostOverviewCard/CostOverviewCard.test.tsx (hunk 2 above)
Comment (acierto) [reply plugins/cost-insights/src/components/CostOverviewCard/CostOverviewCard.test.tsx]: Added the util method in this PR, as it's pretty small and quite fit to this PR as well 😅

**Comment by acierto** [comment]
@jhaals Feedback is incorporated

File: plugins/cost-insights/src/plugin.ts [hunk 3]
```diff
@@ -41,6 +45,14 @@ export const costInsightsPlugin = createPlugin({
     growthAlerts: projectGrowthAlertRef,
     unlabeledDataflowAlerts: unlabeledDataflowAlertRef,
   },
+  __experimentalConfigure(
```
Comment (andrewthauer) [inline plugins/cost-insights/src/plugin.ts:48]: TIL 😄

**Comment by acierto** [comment]
@jhaals moved `MockPluginProvider` to a project wise test-util module

File: .changeset/yellow-forks-knock.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+---
+'@backstage/plugin-cost-insights': patch
+'@backstage/test-utils': patch
```
Comment (jhaals) [inline .changeset/yellow-forks-knock.md:3]: This should be split up into a separate changeset as mentions of cost insights does not need to go into the changelog for test-utils 😄

**Review by jhaals — CHANGES_REQUESTED** [review]
Thanks for the updates! Co-assigning @Rugvip to this PR as he has been involved in the initial work with input types just to make sure that we capture what we need for the test-utils

File: .changeset/yellow-forks-knock.md (hunk 4 above)
Comment (acierto) [reply .changeset/yellow-forks-knock.md]: Oh ok, updated 😄

File: packages/test-utils/api-report.md [hunk 5]
```diff
@@ -165,6 +166,11 @@ export class MockPermissionApi implements PermissionApi {
   ): Promise<EvaluatePermissionResponse>;
 }
 
+// @alpha
+export const MockPluginProvider: ({
```
Comment (Rugvip) [inline packages/test-utils/api-report.md:170]: The `test-utils` package needs to be updated to support the alpha entrypoint. It needs 3 changes in the `package.json`: the added flag to the `build` script, `alpha` in the `files` list, and the entry point added to `publishConfig`.

File: .changeset/gold-yaks-join.md [hunk 6]
```diff
@@ -0,0 +1,6 @@
+---
+'@backstage/test-utils': patch
+---
+
+The test utility for PluginProvider called MockPluginProvider has been created. It will be handy in the cases when you use
+\_\_experimentalConfigure in your plugin
```
Comment (Rugvip) [inline .changeset/gold-yaks-join.md:6]:
```suggestion
The test utility for the plugin context called `MockPluginProvider` has been created. It will be handy in the cases when you use
`__experimentalConfigure` in your plugin. It is experimental and exported through `@backstage/test-utils/alpha`.
```

**Review by Rugvip — CHANGES_REQUESTED** [review]
Thank you! 👍 

Couple of tweaks for the test-util updates.

**Review by jhaals — APPROVED** [review]
This looks good to me 👏 👍

**Review by Rugvip — APPROVED** [review]
👌

**Comment by jhaals** [comment]
Let's :shipit:!
